CoreScriptsRoot/Modules/VR/Dialog.lua

```diff
@@ -53,22 +50,38 @@ end
 function DialogPanel:CalculateTransparency()
 	local headCF = InputService:GetUserCFrame(Enum.UserCFrame.Head)
 	local headLook = headCF.lookVector * Vector3.new(1, 0, 1)
-	local vectorToPanel = Vector3.new(math.cos(dialogPanelAngle + math.rad(90)), 0, -math.sin(dialogPanelAngle + math.rad(90)))
+	local vertAngle = math.asin(headCF.lookVector.Y)
+	local vectorToPanel = Vector3.new(math.cos(baseAngle + dialogPanelAngle + math.rad(90)), 0, -math.sin(baseAngle + dialogPanelAngle + math.rad(90)))
+
+	if math.abs(vertAngle) > PANEL_FADE_ANGLE_1 then
+		resetBaseAngle = true
+	end
 
 	local angleToPanel = math.acos(headLook:Dot(vectorToPanel))
+
+	print(headLook:Dot(vectorToPanel))
```

Comment:
Do you think I could write a small script that automatically removes debug prints for me when I commit?